### DISCUSSION TITLE
Hubs cloud: One more person with SES Email problems

### DISCUSSION THREAD

**Discussion by MegaMotion** [mozilla/hubs#3428] [Hubs Cloud Questions] [answered]
Hi, just wondering if it's a time delay thing or if there is a crucial step I'm missing... I have escaped the AWS sandbox, I have added my base domain name hubsevents.com to SES as a verified domain, and ...? 

Do I need to create an email address, like info@hubsevents.com, for the emails to come from, or does hubs do that automatically? Is there anything else in AWS that I need to turn on before it will be able to send emails? 

It will send email to my admin account but so far (six hours since domain verification) I can't get it to send an email out into the world yet.

**Comment by rawnsley** [comment]
I would start by looking at the SES bounce notifications https://aws.amazon.com/blogs/messaging-and-targeting/handling-bounces-and-complaints/. It's possible it never got as far as SES (although unlikely given that the admin email managed it) so you might also want to check the server logs: https://hubs.mozilla.com/docs/hubs-cloud-accessing-servers.html

**Reply by MegaMotion** [reply]
Okay, thanks! That is helpful. I ran journalctl and learned the following:

```
Nov 25 17:22:50 zealous-mage.hubsevents.com bash[2142]: reticulum.default@hubsevents-3(O): ** (stop) bad return value: {:permanent_failure, "554 Message rejected: Email address is not verified. The following identities failed the check in region US-EAST-1: info@vearth.net\r\n"}
Nov 25 17:22:50 zealous-mage.hubsevents.com bash[2142]: reticulum.default@hubsevents-3(O): Last message: %Phoenix.Socket.Message{event: "auth_request", join_ref: "1", payload: %{"email" => "info@vearth.net", "origin" => "hubs"}, ref: "2", topic: "auth:8a62ec1a-0207-4c4e-9cd6-bd5db8ebc654"}
Nov 25 17:22:50 zealous-mage.hubsevents.com bash[2142]: reticulum.default@hubsevents-3(O): 17:22:50.891 [warn] Failed to send Sentry event.
Nov 25 17:22:50 zealous-mage.hubsevents.com bash[2142]: reticulum.default@hubsevents-3(O): Cannot send event because of invalid DSN
```
I have entered hubsevents.com as a verified domain, but do I need to enter each temporary instance name (eg zealous-mage) on that list as well? Or is the problem that it is still looking for the recipient email addresses to be verified, even though I am supposed to be out of the sandbox by now?

**Reply by wmurphyrd** [reply]
> Email address is not verified. The following identities failed the check in region US-EAST-1: info@vearth.net

That sure looks just like the error you get while you're in sandbox

**Reply by MegaMotion** [reply]
Yup! I finally figured out I was communicating with the wrong region, I'm hosting out of US-WEST-2, but my email server is in US-EAST-1, for whatever reason.

**Comment by BonzoMartini** [comment]
SendGrid emails are blocked as spam at the moment.

**Reply by MegaMotion** [reply]
When I get the admin emails, they do land in my spam folder, but the other email I'm trying to test it on doesn't have a spam folder, just an inbox - do you mean blocked at the AWS end or the receiving end?

**Reply by BonzoMartini** [reply]
Receiving end. Sendgrid seems to be cursed. I don't get an email from SendGrid to my admin address. I can get email from SendGrid to a gmail address. When I look on Sendgrids 'admin' thing I can see that the mails to my admin domain are blocked. And checking with some online blocklist thing says the Sendgrid IP is blocked and their own docs say that this is likely as free Sendgrid emails are all sharing same IPs , so very spammy seeming.

**Reply by BonzoMartini** [reply]
you can write to them to unblock it so sendgrid seems like a total whackamole

**Comment by MegaMotion** [comment, marked as answer]
Aha, the problem appears to be... wait for it... user error! :-) I got out of the sandbox for the US-WEST-2 region where I am hosting, but apparently my mail server is in US-EAST-1 (is that necessary or did I just screw up?) 

I have applied to get out of the sandbox again for US-EAST-1, but at least there I can see my sent mails to the admin account and this seems quite likely to be the root of my problems, lol.

Can anyone tell me if you need to add mail.<yourhubsdomain> to the verified domains list, or if just <yourhubsdomain> will do it?

**Reply by MegaMotion** [reply]
Yup!! This was the problem. Don't forget to check your region! ;-)

**Reply by robinkwilson** [reply]
It will just do it!
